src/ui/component/notifications/notifications.tsx

```diff
@@ -0,0 +1,53 @@
+import type { FC } from 'react'
+import { useMemo, useCallback } from 'react'
+import { useAppStore } from '@/ui/store/appStore'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Toast } from '@/ui/component/toast/toast'
+import type { NotificationType } from '@/domain/notification/entity'
+import { toEffectfulObject } from '@/util/effect'
+import { useNotificationStore } from '@/ui/store'
+import type { DismissNotificationInput } from '@/domain/notification/aggregate'
+
+export const Notifications: FC = () => {
+  const { notifications, dismissNotification } = toEffectfulObject(
+    useNotificationStore(state => ({
+      notifications: state.notifications,
+      dismissNotification: state.dismissNotification
+    }))
+  )
```

Comment:
I would have enhanced the `useDispatchNotification` hook to return the whole store abstraction:

```ts
...
return [notifications, {dispatch, dismiss}]

```

Reply:
Good suggestion. Enhancement has been made in [4b19bdf](https://github.com/okp4/dataverse-portal/pull/195/commits/4b19bdf7d81e3f147407fbf553ede674d564c970).

The implementation was slightly different to your proposal in order to avoid such import:
```tsx
const [, { dispatchNotification }] = useDispatchNotification()
```
or
```tsx
const [{ dispatchNotification }] = useDispatchNotification()
```



With the implementation we now have :
```tsx
const { dispatchNotification } = useDispatchNotification()
```

Is it fine?